### PR TITLE
CLM AddOn: if the user changes the Component.Path, move the directory

### PR DESCRIPTION
**Pull request by ksmyth**

### CODE REVIEW

File: src/ComponentLibraryManagerAddOn/ComponentLibraryManagerAddOn.cs [hunk 1]
```diff
@@ -148,6 +149,71 @@ public void ObjectEvent(MgaObject subject, uint eventMask, object param)
                     Process(CyPhyClasses.ComponentAssembly.Cast(subject));
                 }
             }
+            else if ((eventMask & (uint)objectevent_enum.OBJEVENT_ATTR) != 0 &&
```
Comment (adamnagel) [inline src/ComponentLibraryManagerAddOn/ComponentLibraryManagerAddOn.cs]: I'd like to see a comment or alias for these expressions. For example, above we have `objectCreatedAndNotLibObject`.

**Comment by adamnagel** [comment]
Largely looks good, but I'd like to see an automated test for this. The CLM AddOn should have a complete set of tests to add on to.

**Comment by ksmyth** [comment]
New commit, including automated test. Please have a look

**Comment by adamnagel** [comment]
I've added a test for the case where the "renamed path" already exists. With the current implementation, the copy fails. 

However, my test checks that the path attribute should be unchanged, because my guess is that in this case the rename should be aborted. Thoughts on that?

**Comment by adamnagel** [comment]
Based on our discussion, [ComponentLibraryNamagerAddOn.cs:195](https://github.com/metamorph-inc/meta-core/blob/CLM_Path_Change/src/ComponentLibraryManagerAddOn/ComponentLibraryManagerAddOn.cs#L195) looks like the right place to handle this. However, it gets complicated quickly because [lines 204-211](https://github.com/metamorph-inc/meta-core/blob/CLM_Path_Change/src/ComponentLibraryManagerAddOn/ComponentLibraryManagerAddOn.cs#L204) run immediately after, modifying the path again (for style sanitation) and putting this back on the stack for handling. One potential way to discover this has happened: the path from the second call is semantically the same as the path from the first, and neither match the old path.

**Comment by adamnagel** [comment]
d70dd5c is my attempt to address it.

**Pull request merged**